### PR TITLE
Refactored place detail view, separated image component, re-styled detail view

### PR DESCRIPTION
**Pull request by aej11a**

### CODE REVIEW

File: src/hooks/useFetchPlace.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { SERVER_BASE_URL } from "../util/env";
+import { PlaceDetailsWithAccesibilityData } from "../util/placeTypes";
+
+export const useFetchPlace = ({
+	placeID,
+	includeRatings = true,
+}: {
+	placeID: string;
+	includeRatings?: boolean;
+}) => {
```
Comment (carpenterd777) [inline src/hooks/useFetchPlace.ts:12]: could you add the return type here?

File: src/screens/DetailedViewScreen.tsx [hunk 2]
```diff
@@ -1,75 +1,84 @@
...
+import { getAverageA11yRating } from "../util/processA11yRatings";
 
 export interface PlaceProps {
-	placeID?: string;
-	setPage: () => void;
+	placeID: string;
+	setPage: (newPage: string) => void;
 }
 
+// eslint-disable-next-line
```
Comment (carpenterd777) [inline src/screens/DetailedViewScreen.tsx:17]: could you add the specific rule that we are violating here so that we don't ignore all rules?

**Review by carpenterd777 — CHANGES_REQUESTED** [review]
looks mostly good, just a couple of things I'm worried about that I mentioned in the other review comments.

I see that you moved the server call from `DetailedViewScreen` into its own hook. moving forward, do you think that we should always have server calls isolated to their own hook? that seems reasonable to me

**Comment by aej11a** [comment]
> moving forward, do you think that we should always have server calls isolated to their own hook? that seems reasonable to me

generally yes, there may be some exceptions like if we have a component doing a super specific query that other components would never need, but in most cases i try to separate it out

File: src/screens/DetailedViewScreen.tsx (hunk 2 above)
Comment (aej11a) [reply src/screens/DetailedViewScreen.tsx]: actually was able to remove this, it was a remnant of an intermediate change i removed in the end

File: src/screens/DetailedViewScreen.tsx [hunk 3]
```diff
@@ -1,75 +1,83 @@
...
+					</Text>
+					<BodyText ellipsizeMode="tail" numberOfLines={2}>
+						{place.formatted_address}
+					</BodyText>
+					<BodyText accessibilityLabel={distanceInMi ? `${distanceInMi} miles away` : ""}>
+						{distanceInMi ? `${distanceInMi} mi away` : ""}
+					</BodyText>
+					<BodyText>
+						Accessibility Rating:{" "}
+						{getAverageA11yRating({
```
Comment (carpenterd777) [inline src/screens/DetailedViewScreen.tsx:61]: I noticed that the place list uses `Math.round(getAverageA11yRating(place) * 2) / 2`, while we use this here... I feel like we should change one or the other to be consistent

**Review by carpenterd777 — CHANGES_REQUESTED** [review]
just ran the app for the first time on my phone, adding the image was a good touch! just one more concern

**Review by carpenterd777 — APPROVED** [review]
merging with administrator permission since the codecov CI has yet to be configured to be laxer

**Pull request merged**